### PR TITLE
Launch canonical fast fetch for doubleclick, with holdback

### PR DESCRIPTION
**Pull request by bradfrizzell**
Launch allowing Fast Fetch on non-CDN pages for DoubleClick.

### CODE REVIEW

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js [hunk 1]
```diff
@@ -504,9 +505,9 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
 
   /** @override */
   shouldPreferentialRenderWithoutCrypto() {
-    return experimentFeatureEnabled(
-        this.win, DOUBLECLICK_EXPERIMENT_FEATURE.CANONICAL_EXPERIMENT,
-        DFP_CANONICAL_FF_EXPERIMENT_NAME);
+    return !isCdnProxy(this.win) && !experimentFeatureEnabled(
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: Is this CDN check actually necessary?  If on the CDN, then page is SSL so crypto should be present.  How about instead adding dev().assert(!isCdnProxy(this.win));

File: extensions/amp-ad-network-doubleclick-impl/0.1/doubleclick-a4a-config.js [hunk 2]
```diff
@@ -93,30 +85,29 @@ export const URL_EXPERIMENT_MAPPING = {
 };
 
 /**
+ * Returns whether we are running on the AMP CDN.
+ * @param {!Window} win
+ * @return {boolean}
+ */
+export function isCdnProxy(win) {
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-doubleclick-impl/0.1/doubleclick-a4a-config.js]: Let's put this in ads/google/a4a/utils.js

File: extensions/amp-ad-network-doubleclick-impl/0.1/doubleclick-a4a-config.js [hunk 3]
```diff
@@ -180,32 +171,23 @@ export class DoubleclickA4aEligibility {
...
       if (urlExperimentId == -1 &&
           (getMode(win).localDev || getMode(win).test)) {
         experimentId = MANUAL_EXPERIMENT_ID;
       } else {
         let unconditionedExp;
-        // For unconditioned canonical experiment, in the experiment branch
-        // we allow Fast Fetch on non-CDN pages, but in the control we do not.
+        // For unconditioned canonical holdback, in the control branch
+        // we allow Fast Fetch on non-CDN pages, but in the experiment we do not.
         if ((unconditionedExp = getExperimentBranch(
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-doubleclick-impl/0.1/doubleclick-a4a-config.js]: Can collapse this where we allow FF unless its the holdback exp:

return getExperimentBranch(...) != DOUBLECLICK_UNCONDITIONED_EXPERIMENTS.CANONICAL_HLDBK_EXP;

Comment (bradfrizzell) [reply extensions/amp-ad-network-doubleclick-impl/0.1/doubleclick-a4a-config.js]: done

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js (hunk 1 above)
Comment (bradfrizzell) [reply extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: done

**Comment by bradfrizzell** [comment]
do not merge yet

**Pull request merged**